lib/publiccloud/ec2.pm

```diff
@@ -36,24 +37,27 @@ sub find_img {
     return;
 }
 
+# Returns true if key is already created in EC2 otherwise tries 10 times to create it and then fails
+# If the subroutine manager to create key pair in EC2 it stores it in $self->ssh_key_pair
+
 sub create_keypair {
-    my ($self, $prefix, $out_file) = @_;
+    my ($self, $prefix) = @_;
 
-    return $self->ssh_key_file if ($self->ssh_key_file);
+    return 1 if (script_run('test -f ' . SSH_KEY_PEM) == 0);
```

Comment:
```suggestion
    return 1 if (-s SSH_KEY_PEM);
```
Moreover, shouldn't we set file directory too here ?

Reply:
I'm I don't follow what you're trying to achieve by suggested change.

Reply:
To use perl file check, not script run

Reply:
That won't work @m-dati because `-s` runs the check on the openQA worker, not on the test VM.

Reply:
but it is not ssh_script_run, that as you told, run on VM, but a direct shell run where the same perl code is running: isn't it?

Reply:
`script_run` runs the command in the openQA test wm. `-s SSH_KEY_PEM` runs the command on the worker.

Reply:
anyway also in that case `-s` would be better, to check the file is also not empty

Reply:
Very true mate, please check now :smile: